### PR TITLE
add pdf upload with pdf.js

### PR DESCRIPTION
**Pull request by Uzay-G**
Adds support for uploading PDFs to a data source, using pdf.js

### CODE REVIEW

**Review by spolu — CHANGES_REQUESTED** [review]
1 question and 1 small change :+1:

File: front/pages/w/[wId]/ds/[name]/upsert.tsx [hunk 1]
```diff
@@ -301,10 +330,40 @@ export default function DataSourceUpsert({
                   <h3 className="text-sm font-medium text-gray-700">
                     Text Content
                   </h3>
-                  <p className="my-2 text-sm text-gray-500">
-                    Upload or copy the text data for the document you want to
-                    create or replace (upsert).
-                  </p>
+                  <div className="flex flex-row">
```
[the remark this answers is not in the document]
Comment (Uzay-G) [reply front/pages/w/[wId]/ds/[name]/upsert.tsx:333]: image

[the remark this answers is not in the document]
Comment (spolu) [reply front/pages/w/[wId]/ds/[name]/upsert.tsx:333]: Yep as you can see the upload is not "pixel perfect" :) It's too low and should be vertically centered in the space taken by "Text Content" and they description.

Also can you add "Upload (text or PDF) or copy..." 

Thanks!

**Review by spolu — APPROVED** [review]
:ship:

**Pull request merged**